drivers/sensor/dps310/dps310.c

```diff
@@ -206,11 +206,11 @@ enum {
 #define POW_2_19_MINUS_1	BIT_MASK(19)
 
 /* Needed because the values are referenced by pointer. */
-static const uint8_t REG_ADDR_MEAS_CFG = IFX_DPS310_REG_ADDR_MEAS_CFG;
-static const uint8_t REG_ADDR_CALIB_COEFF_0 = IFX_DPS310_REG_ADDR_COEF_0;
-static const uint8_t REG_ADDR_TMP_B2 = IFX_DPS310_REG_ADDR_TMP_B2;
-static const uint8_t REG_ADDR_PSR_B2 = IFX_DPS310_REG_ADDR_PSR_B2;
-static const uint8_t REG_ADDR_COEF_SRCE = IFX_DPS310_REG_ADDR_COEF_SRCE;
+static uint8_t REG_ADDR_MEAS_CFG = IFX_DPS310_REG_ADDR_MEAS_CFG;
+static uint8_t REG_ADDR_CALIB_COEFF_0 = IFX_DPS310_REG_ADDR_COEF_0;
+static uint8_t REG_ADDR_TMP_B2 = IFX_DPS310_REG_ADDR_TMP_B2;
+static uint8_t REG_ADDR_PSR_B2 = IFX_DPS310_REG_ADDR_PSR_B2;
+static uint8_t REG_ADDR_COEF_SRCE = IFX_DPS310_REG_ADDR_COEF_SRCE;
```

Comment:
We should not be removing `const` qualifier for the immutable variables just because they need to be placed in the RAM region for one of the platforms that Zephyr supports.

The correct solution in this case would be to add a macro like `DPS310_DMA_VAR` that is defined as:

* `__in_section_unique(_DATA_SECTION_NAME)` when compiling for nRF family SoC
* nothing for the rest

Reply:
@stephanosio,

Thanks for your feedback. I updated the PR by declaring a macro `DPS310_DMA_VAR` that complies with leaving `const` immutable variables when not applicable.

LMK what are your thoughts.

Reply:
@stephanosio how do you see this solution scaling across multiple drivers?

Reply:
> @stephanosio how do you see this solution scaling across multiple drivers?

A more fundamental solution would be to define an I2C subsystem-level macro like `I2C_BUF`, but that would be beyond the scope of this PR.